Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,244 @@
+            migrationHandlerTypeNames = migrationTypeNamesList.ToArray();
+
+            selectedMigrationHandlerIndex = 0;
+        }
+
+        private void OnGUI()
+        {
+            using (new EditorGUI.DisabledGroupScope(EditorApplication.isPlaying || EditorApplication.isPaused))
+            {
+                DrawHeader();
+                DrawMigrationTypeSelector();
+
+                using (new EditorGUI.DisabledGroupScope(!isMigrationEnabled))
+                {
+                    DrawMigrationToolbars();
+
+                    if (isMigrationEnabled)
+                    {
+                        DrawObjectSelection();
+                        DrawObjectsForMigration();
+                    }
+                }
+            }
+        }
+
+        private void DrawHeader()
+        {
+            MixedRealityInspectorUtility.RenderMixedRealityToolkitLogo();
+
+            using (new EditorGUILayout.HorizontalScope())
+            {
+                EditorGUILayout.LabelField("Migration Window", EditorStyles.boldLabel);
+                InspectorUIUtility.RenderDocumentationButton(MigrationWindowURL);
+            }
+            EditorGUILayout.LabelField(WindowDescription, EditorStyles.wordWrappedLabel);
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+        }
+
+        private void DrawMigrationTypeSelector()
+        {
+            using (var horizontal = new EditorGUILayout.HorizontalScope())
+            {
+                EditorGUILayout.LabelField("Migration Handler Selection", EditorStyles.boldLabel);
+
+                EditorGUI.BeginChangeCheck();
+
+                selectedMigrationHandlerIndex = EditorGUILayout.Popup(selectedMigrationHandlerIndex, migrationHandlerTypeNames, GUILayout.Width(500));
+                if (EditorGUI.EndChangeCheck())
+                {
+                    SetMigrationHandlerType();
+                }
+            }
+        }
+
+        private void DrawMigrationToolbars()
+        {
+            using (new GUILayout.VerticalScope(EditorStyles.helpBox))
+            {
+                EditorGUILayout.Space();
+
+                using (var check = new EditorGUI.ChangeCheckScope())
+                {
+                    selectedToolbar = (ToolbarOption)GUILayout.Toolbar((int)selectedToolbar, toolbarTitles);
+                    if (check.changed)
+                    {
+                        scrollPosition = Vector2.zero;
+                        migrationTool.ClearMigrationList();
+                    }
+                }
+            }
+        }
+
+        private void DrawObjectSelection()
+        {
+            EditorGUILayout.Space();
+
+            using (new GUILayout.HorizontalScope())
+            {
+                if (selectedToolbar == ToolbarOption.Project)
+                {
+                    if (GUILayout.Button("Add full project for Migration"))
+                    {
+                        migrationTool.AddProjectForMigration();
```

Comment:
Seems like this could take a long time if scanning the whole project. Would be nice to display progress bar to inform user of amount of time left and for option to cancel
https://docs.unity3d.com/ScriptReference/EditorUtility.DisplayProgressBar.html

Look at dependency window for example usage

Reply:
+1 on that! great idea